doc/spec_TransferElement.dox

```diff
@@ -213,9 +232,50 @@ This documnent is currently still **INCOMPLETE**!
+  - 12.5 Adding the same TransferElement twice is not allowed (logic_error).
+  - 12.6 Adding one TransferElement to two different TransferGroups is not allowed (logic_error).
+  - 12.7 Adding TransferElements with AccessMode::wait_for_new_data is not allowed (logic_error).
+  - 12.8 Adding a copy of the a TransferElement which is already in the group (returned by two different calls to Device::getZzzRegisterAccessor() resp. DeviceBackend::getRegisterAccessor() with the same arguments) will replace one of the identical TransferElement with a decorator which copies the data from the other.
+    - 12.8.1 This avoids an unnecessary transfer of identical data.
+    - 12.8.2 This is only possible, if the TransferElement is added by passing a TransferElementAbstractor to TransferGroup::addAccessor().
+  - 12.9 Operations executed via TransferGroup::read() and TransferGroup::write():
+    - 12.9.1 first call preXxx() on all high-level elements (cf. \ref transferElement_B_12_1_2_1 "12.1.2.1"),
+    - 12.9.2 next call the corresponding xxxTransferYyy() on all low-level elements (cf. \ref transferElement_B_12_1_2_2 "12.1.2.2"), and
+    - 12.9.3 finally call postXxx() on all high-level elements.
+  - 12.10 The TransferGroup implements a similar exception handling in TransferGroup::read() and TransferGroup::write() as their TransferElement counterparts. This especially means:
+    - 12.10.1 Before preXxx() is called, the TransferGroup checks all pre-conditions (device opened and operation allowed) to make sure no ChimeraTK::logic_error can occur (other exceptions are anyway not allowed in preXxx).
+      - \anchor transferElement_B_12_10_1_1 12.10.1.1 If any ChimeraTK::runtime_error occurs while checking the pre-conditions, it is immediately passed on to the application.
+      - \anchor transferElement_B_12_10_1_2 12.10.1.2 Any pre-condition which is not met will immediately result in a thrown ChimeraTK::logic_error
+      - 12.10.1.3 The exceptions from \ref transferElement_B_12_10_1_1 "12.10.1.1" and \ref transferElement_B_12_10_1_2 "12.10.1.2" are not prioritised by type. The first detected error throws.
+    - 12.10.2 ChimeraTK::runtime_error exceptions thrown by the low-level elements are propagated to the corresponding high-level elements so they are seen by their postXxx() functions (c.f. \ref transferElement_B_6_3 "6.3")
+    - 12.10.3 postXxx() is called for all high level elements, even if some postXxx() throw exceptions (c.f. \ref transferElement_B_5 "5").
+      - 12.10.3.1 If exceptions have been caught from any postXxx(), an exception is thrown after all of them have been executed.
+      - 12.10.3.2 If more than one exception type is detected, the exception type with the highest priority is thrown:
+        - ChimeraTK::logic_error is thrown before any transfer is actually started, so it implicitly has highest priority.
+        - If a ChimeraTK::runtime_error is thrown by a transfer, it will be passed on to the application.
```

Comment:
This is confusing. And there can be multiple. I will propose something below which hopefully is clearer.

Reply:
Only one exception can be thrown at a time in C++. I want to avoid the need for a sophisticated to prioritise exception types at that point, because there is no benefit from that. You can always have first a runtime_error, because no communication is possible, and when this is resolved a logic_error, because only then the information to decide about the logic_error is available. Keep in mind, the TransferGroup does not reveal which accessor has thrown.

Reply:
> because only then the information to decide about the logic_error is available

I don't think the information about logic errors can be done after a transfer has thrown a runtime_error. We are not talking about the runtime_errors in the checking pre-conditions phase. Those have been dealt with at this point. It's only about exceptions from the transfer phase.

Reply:
Ah, now I know what you mean. In the next call to read. But that's a new round.

Reply:
Ok I think I don't understand what your comment refers to here. Is this all maybe already discussed below in the other points? I am lost.

Reply:
Let's converge this discussion in the thread about 12.10.3.1.